content/en/blog/_posts/2023-09-15-Bootstrap-an-Air-Gapped-Cluster-with-Kubeadm/index.md

```diff
@@ -0,0 +1,665 @@
+# Since this is a single node cluster, this taint has to be commented out,
+# otherwise the coredns pods will not come up.
+# taints:
+# - effect: NoSchedule
+# key: node-role.kubernetes.io/master
+bootstrapTokens:
+- groups:
+    - system:bootstrappers:kubeadm:default-node-token
+    token: abcdef.0123456789abcdef
+    ttl: 24h0m0s
+    usages:
+    - signing
+    - authentication" > kubeadm_cluster.yaml
+
+kubeadm init --config kubeadm_config.yaml
+```
+
+**Export the KUBECONFIG and wait until the API server is healthy:**
+
+```bash
+export KUBECONFIG=/etc/kubernetes/admin.conf
+
+until kubectl get nodes; do
+    echo -e "\nWaiting for API server to respond..."
+    sleep 5
+
+done
+```
+
+**Update Flannel image locations in the Flannel manifest and apply:**
+
+```bash
+sed -i 's/image: docker\.io/image: localhost:5000/g' kube-flannel.yaml
+kubectl apply -f kube-flannel.yaml
+```
+
+Run `kubectl get pods -A --watch` until all pods are up and running.
+
+# Run an example Pod
+
+With a cluster operational, the next step is a workload. For this simple demonstration, the [Podinfo](https://github.com/stefanprodan/podinfo) application will be deployed.
+
+## Helm
+
+This first part of the procedure must be executed from the host/laptop. If not already present, install Helm following [Installing Helm](https://helm.sh/docs/intro/install/).
+
+Next, download the helm binary for Linux:
+
+```bash
+UARCH=$(uname -m)
+# Reset the architecture variables if needed
+if [["$UARCH" == "arm64" || "$UARCH" == "aarch64"]]; then
+
+    ARCH="aarch64"
+    K8s_ARCH="arm64"
+
+else
+
+    ARCH="x86_64"
+    K8s_ARCH="amd64"
+
+fi
+
+curl -LO https://get.helm.sh/helm-v3.12.2-linux-${K8s_ARCH}.tar.gz
+```
+
+Add the Podinfo helm repository, download the Podinfo helm chart, download the Podinfo container image, and then finally save it to the local disk:
+
+```bash
+helm repo add https://stefanprodan.github.io/podinfo
+helm fetch podinfo/podinfo --version 6.4.0
+docker pull ghcr.io/stefanprodan/podinfo:6.4.0
+```
+
+**Save the image to a tar file on the local disk**
+```bash
+docker save -o podinfo_podinfo-6.4.0.tar ghcr.io/stefanprodan/podinfo
+```
+**Reuse the ~/tmp directory created on the air gapped VM to transport these artifacts across the air gap:**
+```bash
+scp -i <<SSH_KEY>> <<FILE>> <<AIRGAP_VM_USER>>@<<AIRGAP_VM_IP>>:~/tmp/
+```
+Now pivot over to the air gap VM for the rest of the installation procedure.
+
+Switch into ~/tmp:
+```
+cd ~/tmp
+```
+Extract and move the helm binary:
+
+```
+tar -zxvf helm-v3.0.0-linux-amd64.tar.gz
+mv linux-amd64/helm /usr/local/bin/helm
+```
+
+**Load the Podinfo container image into the local Docker registry:**
+
+```bash
+docker load -i podinfo_podinfo-6.4.0.tar
+docker tag podinfo/podinfo:6.4.0 localhost:5000/podinfo/podinfo:6.4.0
+docker push localhost:5000/podinfo/podinfo:6.4.0
+```
+
+**Ensure the proper KUBECONFIG is set and install the Podinfo helm chart:**
+
+```
```

Comment:
Mark this as shell?

Reply:
@sftim , I'll go back through and update all pertinent code blocks to be bash/shell/yaml/etc as appropriate.